magit-gh-comments.el

```diff
@@ -187,25 +305,155 @@ Github-style position."
        gh-hunk-offset)))
 
 
+(defun magit-gh--diff-pos/gh->magit (file gh-pos gh-diff-body)
+  "Convert a Github diff position to a Magit diff position.
+
+Given a FILE and a Github-style diff position GH-POS in
+GH-DIFF-BODY, return the corresponding magit-gh-diff-pos."
+  (with-temp-buffer
+    (save-excursion
+      (insert gh-diff-body)
+      (goto-char (point-min))
+      (re-search-forward (format "+++ b/%s" file))
+      (re-search-forward "@@ -\\([0-9]+\\),\\([0-9]+\\) \\+\\([0-9]+\\),\\([0-9]+\\) @@")
+      (forward-line gh-pos)
+      (let ((rev (if (looking-at "^-") :a :b))
+            (offset 0))
+        (while (not (magit-gh--try-parse-hunk-header))
+          (assert (= 0 (forward-line -1)))
+          ;; (beginning-of-line)
+          (unless (or (and (eq rev :a) (looking-at "^\\+"))
+                      (and (eq rev :b) (looking-at "^-")))
+            (cl-incf offset)))
+        (make-magit-gh-diff-pos :a-or-b rev
+                                :hunk-start (alist-get
+                                             (if (eq rev :a) :a-start :b-start)
+                                             (magit-gh--try-parse-hunk-header))
+                                :offset (1- offset))))))
+
+
 (defun magit-gh--current-section-content ()
-  "Return the contents of the magit section at point as a string"
+  "Return the contents of the magit section at point."
   (interactive)
   (buffer-substring-no-properties
    (magit-section-start (magit-current-section))
    (magit-section-end (magit-current-section))))
 
 
+;;;###autoload
 (defun magit-gh-add-comment (comment-text)
-  "Comment on the line at point and post the comment to Github"
+  "Comment on the line at point and post COMMENT-TEXT to Github."
   (interactive "MComment: ")
-  (let* ((lines (split-string (magit-gh--current-section-content) "\n" t))
-         (magit-pos (magit-gh--cur-diff-pos))
-         (github-pos (magit-gh--translate-diff-pos magit-pos (magit-gh--current-section-content)))
-         (commit-sha (magit-diff-visit--range-end)))
-    (magit-gh--post-pr-comment magit-gh-comment-test-pr
-                               (magit-current-file)
+  (let* ((magit-pos (magit-gh--cur-diff-pos)) ;; TODO: What if the line at point isn't part of a hunk/isn't changed by this diff?
+         (diff-body (magit-gh--fetch-diff-from-github (magit-gh--get-current-pr)))
+         (github-pos (magit-gh--diff-pos/magit->gh (magit-current-file) magit-pos diff-body))
+         (commit-sha (cdr (magit-split-range (magit-diff--dwim)))))
```

Comment:
I think these comments are consistently off-by-one. For example, my point in Emacs is on the line that begins '(github-pos ', but I think it's going to be posted to Github with a line number pointing to the line that begins with '(commit-sha '